core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2632,6 +2642,22 @@ public void updateMount(AlluxioURI alluxioPath, MountContext context)
           throw t;
         }
       }
+      if (syncPoints != null && !syncPoints.isEmpty()) {
+        // add back all sync points
+        Exception syncEx = null;
+        for (AlluxioURI syncPoint : syncPoints) {
+          try {
+            startSyncAndJournal(rpcContext, syncPoint);
```

Comment:
Is this equivalent to keeping the sync point, or will we incur a large cost of restarting the sync (same as if you needed to add a mount as a new sync point)

Reply:
There will be some cost equivalent to adding the sync points again, but that is necessary because we have to reinitiate the sync hook using the new UFS client.